### PR TITLE
Display hints: Ready for merge

### PR DESCRIPTION
**Pull request by ratulm**

### CODE REVIEW

File: tests/jsonpath-addons/jsonpath-display-hints-prefixofsuffix.ref [hunk 1]
```diff
@@ -0,0 +1,1221 @@
+{
+  "answerElements" : [
+    {
+      "class" : "org.batfish.question.jsonpath.JsonPathQuestionPlugin$JsonPathAnswerElement",
+      "results" : {
+        "0" : {
+          "displayValues" : {
+            "'nodes'->'as2border1'" : {
```
Comment (dhalperi) [inline tests/jsonpath-addons/jsonpath-display-hints-prefixofsuffix.ref]: I don't understand how this version of display values is useful. Given this and the information in the displayHints structure, what can I do with it?

I don't know the type of the key or the values, I don't have a text thing to put them in, etc.

File: projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathDisplayHintsTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    Map<String, Map<String, JsonNode>> displayValues =
+        result.computeDisplayValues(getDisplayHints(displayVariable));
+
+    String key1 = "'nodes'->'node1'->'interfaces'";
+    String key2 = "'nodes'->'node2'->'interfaces'";
+
+    assertThat(displayValues.size(), equalTo(2));
+    assertThat(displayValues.containsKey(key1), equalTo(true));
+    assertThat(displayValues.containsKey(key2), equalTo(true));
```
Comment (dhalperi) [inline projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathDisplayHintsTest.java]: shouldn't the key here be "interfaceListFromPrefixOfSuffix"? I thought it was taken from the name of the extraction hint, like `node1` in displayHintsTest.json

File: projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathResult.java [hunk 3]
```diff
@@ -68,8 +104,151 @@ public int hashCode() {
...
   }
 
+  public static class JsonPathExtractionHint {
+
+    public enum UseType {
+      PREFIX("prefix"),
+      PREFIXOFSUFFIX("prefixofsuffix"),
+      SUFFIXOFSUFFIX("suffixofsuffix");
+
+      private static final Map<String, UseType> MAP = initMap();
```
[the remark this answers is not in the document]
Comment (ratulm) [reply projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathResult.java]: Yes, this type of construction is used all over the code, to make enums case-insensitive.

[the remark this answers is not in the document]
Comment (dhalperi) [reply projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathResult.java]: ack, but why?

File: projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathResult.java [hunk 4]
```diff
@@ -68,8 +104,151 @@ public int hashCode() {
     }
   }
 
+  public static class JsonPathExtractionHint {
```
[the remark this answers is not in the document]
Comment (ratulm) [reply projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathResult.java]: made its own class.

File: projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathAssertionTest.java [hunk 5]
```diff
@@ -48,9 +43,10 @@
 
   @Test
   public void testEvaluateCountFalse() {
-    Set<JsonPathResultEntry> results =
-        computeResults(
-            "org/batfish/question/jsonPathAssertionTest.json", "$.nodes..interface1.mtu", true);
+    Set<JsonPathResultEntry> results = computeResults(
+        "org/batfish/question/jsonpath/jsonPathAssertionTest.json",
+        "$.nodes..interface1.mtu",
+        true);
```
[the remark this answers is not in the document]
Comment (ratulm) [reply projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathAssertionTest.java]: the auto-formatter is on but it is creating lines longer than 100 columns for me. i'll show you tomorrow.

File: projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathDisplayHintsTest.java (hunk 2 above)
Comment (ratulm) [reply projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathDisplayHintsTest.java]: you are likely misunderstanding the data structure layout. more tomorrow.

File: projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathDisplayHintsTest.java [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        computeJsonPathResult(
+            "org/batfish/question/jsonpath/jsonPathDisplayHintsTestObject.json",
+            "$.nodes[*].interfaces[*][?(@.mtu < 1600)].mtu", false);
+
+    String errorMessage = "Cannot compute suffix-based display values with null suffix. "
+        + "(Was suffix set to True in the original JsonPath Query?)";
+    _thrown.expect(BatfishException.class);
+    _thrown.expectMessage(errorMessage);
+
+    Map<String, Map<String, JsonNode>> displayValues =
```
[the remark this answers is not in the document]
Comment (ratulm) [reply projects/question/src/test/java/org/batfish/question/jsonpath/JsonPathDisplayHintsTest.java]: done.

File: tests/jsonpath-addons/jsonpath-display-hints-prefixofsuffix.ref (hunk 1 above)
Comment (ratulm) [reply tests/jsonpath-addons/jsonpath-display-hints-prefixofsuffix.ref]: yeah, it is not complete from usability perspective. more tomorrow.

File: projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathResult.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (dhalperi) [reply projects/question/src/main/java/org/batfish/question/jsonpath/JsonPathResult.java]: oh, case-*insensitive*. Duh, I'm dumb.

They added a default option for that in 2.9.0, we should turn it on and delete all this code (later).

**Pull request merged**